robot_framework/eflyt.py

```diff
@@ -0,0 +1,263 @@
+"""This module contains all logic related to the Eflyt system."""
+
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.select import Select
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.common.exceptions import TimeoutException
+from OpenOrchestrator.orchestrator_connection.connection import OrchestratorConnection
+from OpenOrchestrator.database.queues import QueueStatus
+from itk_dev_shared_components.eflyt import eflyt_case, eflyt_search
+from itk_dev_shared_components.eflyt.eflyt_case import Case
+from robot_framework import config, letters
+
+
+def filter_cases(cases: list[Case], previous_case_numbers: list[str]) -> list[Case]:
+    """Filter cases from the case table.
+
+    Args:
+        cases: A list of cases to filter.
+
+    Returns:
+        A list of filtered case objects.
+    """
+    case_types_to_skip = (
+        "Børneflytning 1",
+        "Børneflytning 2",
+        "Børneflytning 3",
+        "Barn",
+        "Mindreårig",
+        "Sommerhus",
+        "Nordisk land"
+    )
+
+    filtered_cases = [
+        case for case in cases
+        if "Logivært" in case.case_types and case.case_number not in previous_case_numbers
+        and not any(case_type in case_types_to_skip for case_type in case.case_types)
+    ]
+
+    return filtered_cases
+
+
+def handle_case(browser: webdriver.Chrome, case: Case, orchestrator_connection: OrchestratorConnection) -> None:
+    """Handle a single case with all steps included.
+
+    Args:
+        browser: The webdriver browser object.
+        case: The case to handle.
+        orchestrator_connection: The connection to Orchestrator.
+    """
+    if not check_queue(case, orchestrator_connection):
+        return
+
+    # Create a queue element to indicate the case is being handled
+    queue_element = orchestrator_connection.create_queue_element(config.QUEUE_NAME, reference=case.case_number)
+    orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.IN_PROGRESS)
+
+    orchestrator_connection.log_info(f"Beginning case: {case.case_number}")
+
+    eflyt_search.open_case(browser, case.case_number)
+    if not check_followup(browser):
+        orchestrator_connection.set_queue_element_status(queue_element.id, QueueStatus.DONE, message="Sprunget over fordi breve ikke består af en enkelt logiværtserklæring.")
+        orchestrator_connection.log_info("Skipping: Number of letters on case.")
+        return
+
+    if send_letter_to_anmelder(browser, letters.LETTER_TITLE):
```

Comment:
The letter title here should be the title of the letter that was previously sent. Not a hard coded value